jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -1239,4 +1359,379 @@ class Endpoint @JvmOverloads constructor(
 
         override fun trace(f: () -> Unit) = f.invoke()
     }
+
+    /**
+     * Align common fields from different source types.
+     * Perhaps this could become a base class of those types.
+     */
+    private class SourceDesc private constructor(
```

Comment:
Is it possible to move the source remapping code out of Endpoint.kt?  It seems like it would make it easier to comprehend, and also (if you abstract its connection to Endpoint, or mock something) it could make it easier to write unit tests for it.

Reply:
done. yes, this should let me overcome the issues with my unit test. i will update the test and submit it.